### PR TITLE
replaced CGLIB with ByteBuddy 

### PR DESCRIPTION
**Pull request by hamid646m**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -206,22 +208,22 @@
...
-        </configuration>
-        <executions>
-          <execution>
-            <goals>
-              <goal>format</goal>
-            </goals>
-          </execution>
-        </executions>
-      </plugin>
+<!--      <plugin>-->
```
Comment (badgerwithagun) [inline pom.xml:211]: Commented out?

Comment (hamid646m) [inline pom.xml:211]: https://github.com/coveooss/fmt-maven-plugin/issues/99

not compatible with java 17 yet

**Review by badgerwithagun — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (hamid646m) [reply pom.xml:211]: this plugin is not compatible with java 17 https://github.com/coveooss/fmt-maven-plugin/issues/99. should i remove it?

Comment (badgerwithagun) [reply pom.xml:211]: It's possible to override behaviour for specific java versions in maven:
```
<profile>
    <id>active-on-jdk-11</id>
    <activation>
        <jdk>11</jdk>
    </activation>
</profile>
```
So you can enable this plugin for all java versions prior to 17

Comment (badgerwithagun) [reply pom.xml:211]: It's possible to override behaviour for specific java versions in maven:
```
<profile>
    <id>active-on-jdk-11</id>
    <activation>
        <jdk>11</jdk>
    </activation>
</profile>
```
So you can enable this plugin for all java versions prior to 17

Comment (hamid646m) [reply pom.xml:211]: cool, i will do that

File: transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultInvocationSerializer.java [hunk 2]
```diff
@@ -329,6 +338,112 @@ private String nameForClass(Class<?> clazz) {
     }
   }
 
+  static final class LocalDateTimeTypeAdapter extends TypeAdapter<LocalDateTime> {
+
+    @Override
+    public void write(JsonWriter out, LocalDateTime value) throws IOException {
+      out.value(value.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME));
```
Comment (badgerwithagun) [inline transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultInvocationSerializer.java:345]: Should these have the null checks used in `UtcTypeAdapter`?

File: transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultInvocationSerializer.java [hunk 3]
```diff
@@ -329,6 +338,112 @@ private String nameForClass(Class<?> clazz) {
...
+      out.value(value.format(formatter));
+    }
+
+    @Override
+    public MonthDay read(JsonReader in) throws IOException {
+      return MonthDay.parse(in.nextString(), formatter);
+    }
+  }
+
+  static final class PeriodTypeAdapter extends TypeAdapter<Period> {
```
Comment (badgerwithagun) [inline transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultInvocationSerializer.java:408]: A few of these are basically identical and could just be a `ToStringTypeAdapter`

**Review by badgerwithagun — CHANGES_REQUESTED** [review]
Looking great. Just a couple of small suggestions

File: transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultInvocationSerializer.java (hunk 2 above)
Comment (hamid646m) [reply transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultInvocationSerializer.java:345]: nor required, added a test for null

File: transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultInvocationSerializer.java (hunk 3 above)
Comment (hamid646m) [reply transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/DefaultInvocationSerializer.java:408]: but `read` method  won't be easy to implement

**Review by badgerwithagun — APPROVED** [review]

**Pull request merged**